libs/martech/martech.js

```diff
@@ -97,49 +103,93 @@ const getDtmLib = (env) => ({
       : env.consumer?.marTechUrl || 'https://assets.adobedtm.com/d4d114c60e50/a0e989131fd5/launch-a27b33fc2dc0-development.min.js',
 });
 
-export default async function init({ persEnabled = false, persManifests }) {
+const setupEntitlementCallback = () => {
+  const setEntitlements = async (destinations) => {
+    const { default: parseEntitlements } = await import('../features/personalization/entitlements.js');
+    return parseEntitlements(destinations);
+  };
+
+  const getEntitlements = (resolve) => {
+    const handleEntitlements = (detail) => {
+      if (detail?.result?.destinations?.length) {
+        resolve(setEntitlements(detail.result.destinations));
+      } else {
+        resolve([]);
+      }
+    };
+    waitForEventOrTimeout(ALLOY_SEND_EVENT, ENTITLEMENT_TIMEOUT, [])
+      .then(handleEntitlements);
```

Comment:
If `waitForEventOrTimeout` gets rejected (https://github.com/adobecom/milo/pull/1649/files#diff-01f2571b06af2ef537675645367e6758c40e1d7baec6d783e6c6ac9512119a35R26) then `handleEntitlements` is not executed anymore and `getEntitlements` would not get resolved.

Reply:
👍 added catch